.github/workflows/cloud_build_failure_reporter.yml

```diff
@@ -0,0 +1,168 @@
+                        title: title,
+                        body: txt,
+                        labels: [periodicLabel]
+                      });
+                      return;
+                    }
+                    if (prevIssues.length > 1) {
+                      console.warn(
+                        `found ${prevIssues.length} issues but only adding comment to ${prevIssues[0].html_url}`
+                      );
+                    }
+                    console.log(
+                      `found previous issue ${prevIssues[0].html_url}, adding comment`
+                    );
+                    await github.rest.issues.createComment({
+                      ...context.repo,
+                      issue_number: prevIssues[0].number,
+                      body: txt
+                    });
+                  };
+
+                  // updateIssues comments on any existing issues. No-op if no issue exists.
+                  const updateIssues = async function (txt) {
+                    if (prevIssues.length < 1) {
+                      console.log('no previous issues found.');
+                      return;
+                    }
+                    for (const prevIssue of prevIssues) {
+                      console.log(`found previous issue ${prevIssue.html_url}, adding comment`);
+                      await github.rest.issues.createComment({
+                        ...context.repo,
+                        issue_number: prevIssue.number,
+                        body: txt
+                      });
+                    }
+                  };
+
+                  // Find status of check runs.
+                  // We will find check runs for each commit and then filter for the periodic.
+                  // Checks API only allows for ref and if we use main there could be edge cases where
+                  // the check run happened on a SHA that is different from head.
+                  const commits = await github.paginate(github.rest.repos.listCommits, {
+                    ...context.repo
+                  });
+
+                  const relevantChecks = [];
+                  for (const commit of commits) {
+                    console.log(
+                      `checking runs at ${commit.html_url}: ${commit.commit.message}`
+                    );
+                    const checks = await github.rest.checks.listForRef({
+                      ...context.repo,
+                      ref: commit.sha
+                    });
+
+                    // Iterate through each check and find matching names
+                    for (const check of checks.data.check_runs) {
+                      for (const testName of testNameFound.keys()) {
+                        if (testNameFound.get(testName) === true){
+                          //skip if a check is already found for this name
+                          continue; 
+                        }
+                        if (check.name == testName) {
```

Comment:
```suggestion
                        if (check.name.includes(testName)) {
```

Reply:
We will have to ask the user to provider complete check names because we will stop looking for tests once the check name is found. So even if the user provides a substring such as "nightly" and there are multiple triggers with "nightly' substring, the reporter will only find whatever trigger that has the latest run and then stop looking for the rest.

Reply:
Update: still changed to includes because cloud build automatically append project name to trigger names. But we should make it clear to the users that they are supposed to use complete trigger names.